kratos/utilities/divide_geometry.h

```diff
@@ -0,0 +1,331 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_DIVIDE_GEOMETRY)
+#define KRATOS_DIVIDE_GEOMETRY
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/node.h"
+#include "geometries/point.h"
+#include "geometries/geometry.h"
+#include "geometries/geometry_data.h"
+#include "utilities/indexed_object.h"
+#include "containers/pointer_vector_set.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+class KRATOS_API(KRATOS_CORE) IndexedPoint : public Point, public IndexedObject
```

Comment:
It may be interesting to put this on a header on its own. @pooyan-dadvand shall we make it as a first citizen of kratos (read, with his own header and cpp) or consider it as an implementation detail as it is now?